code/game/machinery/sci_bombardment.dm

```diff
@@ -0,0 +1,257 @@
+		to_chat(usr, "<span class='warning'>[B] is refused, as it is invalid or incomplete.</span>")
+	return
+
+/**
+* Starts countdown sequence for firing the TTV
+* 
+* Subtracts 1 from var/tick every second silently until reaching
+* the last 5 seconds. Spawn(10) loops back to the beginning by
+* triggering the proc again after 10 deciseconds. 
+*/
+/obj/machinery/sci_bombardment/proc/countdown()
+	if(stopcount) //Abort launch
+		tick = countdown
+		return
+	tick -= 1
+	if(tick == 0)
+		stopcount = TRUE
+		tick = countdown
+		fire_ttv()
+		radio.talk_into(src, "Payload successfully deployed.",)
+		use_power = 1
+		return
+	else if(tick <= 5)
+		use_power = 2
+		radio.talk_into(src, "[tick]...",)
+	playsound(src, 'sound/effects/clock_tick.ogg', 80, 0)
+	spawn(10)
+		countdown()
+
+/**
+* Launches TTV from LAM to turf
+* 
+* Triggered after being called in the last step of countdown().
+* Sends the loaded TTV to the Turf selected during ui_act("target"),
+* triggers toggle_valve(), and last resets variables to initial state.
+*/
+/obj/machinery/sci_bombardment/proc/fire_ttv()
+	if(!scibomb || !dest)
+		return
+	playsound(src, 'sound/effects/gravhit.ogg', 80, 0.25)
+	playsound(src, 'sound/effects/podwoosh.ogg', 80, 0.25)
+	scibomb.forceMove(dest)
+	playsound(scibomb, 'sound/effects/bamf.ogg', 95, 0.25, 75, 1, 0, 0, FALSE, TRUE) //Minimum impact sound in the odd event Toxins doesn't send a bomb
+	scibomb.toggle_valve()
+	dest = initial(dest)
+	targetdest = initial(dest)
+	tcoords = initial(tcoords)
+	scibomb = initial(scibomb)
+	update_icon()
+	. = TRUE
+
+/**
+* Visual proc, temporarily disables interacting
+* 
+* Triggered after the ui_act's "target" and "abort" to
+* add a delay between selecting GPS coordinates and 
+* cancelling a TTV launch.
+*/
+/obj/machinery/sci_bombardment/proc/reset_lam() 
+	target_delay = !target_delay
+	update_icon()
+	if(target_delay)
+		spawn(100)
+			reset_lam()
+	else
+		playsound(src, 'sound/items/scanner_match.ogg', 100, 1)
+		return
+
+/obj/machinery/sci_bombardment/ui_interact(mob/user, ui_key = "lam", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		var/lam_window_height = clamp(325 + GLOB.GPS_list.len * 24, 325, 700)
+		ui = new(user, src, ui_key, "Lam", name, 470, lam_window_height, master_ui, state)
+		ui.open()
+
+/obj/machinery/sci_bombardment/ui_data(mob/user)
+	var/list/data = list()
+	data["tcoords"] = tcoords
+	data["countdown"] = countdown
+	data["locked"] = locked
+	data["loaded"] = scibomb
+	data["stopcount"] = stopcount
+	if(locked)
+		return data
+	if(tcoords)
+		data["tcoords"] = tcoords
+		data["targetdest"] = targetdest
+
+	var/list/signals = list()
+	data["signals"] = list()
+
+	for(var/gps in GLOB.GPS_list)
+		var/obj/item/gps/G = gps
+		var/turf/pos = get_turf_global(G) // yogs - get_turf_global instead of get_turf
+		if(G.emped || !G.tracking || pos.z != 5)
```

Comment:
use the define for the z level and not a hard constant for maintainability please